### ISSUE TITLE
Does _msize() return the amount of memory that was actually allocated, or that which was requested?

### ISSUE THREAD

**Issue by swineone** [MicrosoftDocs/cpp-docs#482]
I'm unit-testing some code which calls realloc(). I want to test whether a certain buffer was actually realloc()ed with the correct size. _msize() looks perfect for this, except for one issue.

The documentation for malloc() claims that "The malloc function allocates a memory block of at least size bytes." -- emphasis on "at least".

Therefore, there are two reasonable behaviors for _msize(): either it returns the exact value that was passed to malloc(), or it returns the value that was actually allocated, which may be larger than what was requested.

If the actual behavior was the first one, then _msize() would be guaranteed to work in my unit tests. However, if the actual behavior was the second one, my tests might fail (or, at least, I'd have to figure out alignment requirements).

The documentation for _msize() is unclear on which of the two behaviors are actually implemented. I request that this be clarified in the document so that I know whether I can use _msize() in my unit tests.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: f7c33f57-8754-6b50-905e-803038d129b3
* Version Independent ID: 677fd117-6dea-268a-6264-41386b66316a
* Content: [_msize](https://docs.microsoft.com/en-us/cpp/c-runtime-library/reference/msize?view=vs-2017)
* Content Source: [docs/c-runtime-library/reference/msize.md](https://github.com/Microsoft/cpp-docs/blob/master/docs/c-runtime-library/reference/msize.md)
* Product: **visual-cpp**
* GitHub Login: @corob-msft
* Microsoft Alias: **corob**

**Comment by corob-msft** [comment]
@swineone If you look at the UCRT source code in the Windows SDK (on my machine, this is in C:\Program Files (x86)\Windows Kits\10\Source\10.0.17134.0\ucrt) you'll find the retail build's version of `_msize` just invokes the Win32 [HeapSize](https://docs.microsoft.com/en-us/windows/desktop/api/heapapi/nf-heapapi-heapsize) function. The description there is the same as in the `_msize` topic. Similarly, if you look at how `malloc` works, the underlying Win32 [HeapAlloc](https://docs.microsoft.com/en-us/windows/desktop/api/heapapi/nf-heapapi-heapalloc) function also says it returns at least the size requested, but possibly more. I tried a few odd sizes with this code, and always got back the same size I asked for:

```
// _msize.cpp
// compile with: cl /W4 /EHsc /nologo _msize.cpp
#include <iostream>
#include <malloc.h>

int main()
{
    char* pchar = static_cast<char *>(malloc(1048577));
    std::cout << "Allocated " << _msize(pchar) << std::endl;
}
```

The bottom line is, the `_msize` documentation reflects what the documentation for the underlying implementing function says, and we can't differ from that without risk of being wrong.
